application/src/main/scala/com/azavea/franklin/api/services/CollectionItemsService.scala

```diff
@@ -173,17 +173,21 @@ class CollectionItemsService[F[_]: Sync](
       case None =>
         val withParent = item.copy(links = fallbackCollectionLink +: item.links)
         StacItemDao.insertStacItem(withParent).transact(xa) map { inserted =>
-          Right((inserted.asJson, inserted.##.toString))
+          val validated = validateItemAndLinks(inserted)
+          Right((validated.asJson, validated.##.toString))
         }
     }
   }
 
   def putItem(
-      collectionId: String,
-      itemId: String,
+      rawCollectionId: String,
+      rawItemId: String,
       itemUpdate: StacItem,
       etag: String
-  ): F[Either[CrudError, (Json, String)]] =
+  ): F[Either[CrudError, (Json, String)]] = {
+    val itemId       = URLDecoder.decode(rawItemId, StandardCharsets.UTF_8.toString)
```

Comment:
That's a really good id -- I'm thinking `URLSafeItemId` and `URLSafeCollectionId`, and the codec will look a lot like what the [`PaginationToken` codec](https://github.com/azavea/franklin/blob/9c151ede28c8b646f961c0812e925bc03aa7e084/application/src/main/scala/com/azavea/franklin/api/schemas/package.scala#L84-L85) looked like, but with url decoding instead of base64. I'll add that in a sec

Reply:
Oof... this is resulting in a lot of changes. I'm going to open an issue and throw a `1` on it instead.

Reply:
#308